### ISSUE TITLE
Indexing of Docs only able to grab entrypoint page

### ISSUE THREAD

**Issue by JarradJarrad** [getcursor/cursor#1479]
1) If you can, please include a screenshot of your problem

When attempting to index DSPy docs via the chat window, I get this:
![image](https://github.com/getcursor/cursor/assets/41174864/7c07ccbe-8197-4b79-a36e-b59a8e383335)

When attempting to index the same docs via cursor settings -> features -> docs section, I get this: 

![image](https://github.com/getcursor/cursor/assets/41174864/db536aa4-5ad4-4b30-9aa0-715a40061b0d)

I have tried all of the below with the github repo the docs are built from, same thing happens.

2) Please include the name of your operating system

Version: 0.33.4
VSCode Version: 1.86.2
Commit: 2dd5b2855ac68f3a3d3babee87d22b399a9db7d0
Date: 2024-05-11T02:05:08.197Z
Electron: 27.2.3
ElectronBuildId: undefined
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.1.0

3) If you can, steps to reproduce are super helpful

1. navigate to cursor settings -> features -> docs section
2. click 'add new doc'
3. enter the following (with or without /docs/ and with or without trailing slashes, it all results in the same outcome)
![image](https://github.com/getcursor/cursor/assets/41174864/40a57bd8-d2e8-4809-82f3-6833c91567cb)

4. wait while Cursor indexes the docs
5. all it indexes is actually just the entrypoint page

**Comment by jamesryancooper** [comment]
I've had the same issue with numerous docs.

**Comment by jason-downs** [comment]
Yup, this is making the custom docs feature almost useless for me.

**Comment by davidalpha1** [comment]
Any update on this feature being able to navigate sitemaps better?

Tried to use [](https://www.xml-sitemaps.com/), but cursor just indexes the entry page.

**Comment by fmontes** [comment]
I have the same problem with https://dev.dotcms.com/docs/table-of-contents, not even sending the sitemap directly like https://dev.dotcms.com/sitemap.xml

**Comment by kristojorg** [comment]
There are now many issues related to this, none of which have received information from the team :/. 

- https://github.com/getcursor/cursor/issues/2750
- https://github.com/getcursor/cursor/issues/2749
- https://github.com/getcursor/cursor/issues/2488
- https://github.com/getcursor/cursor/issues/2034
- https://github.com/getcursor/cursor/issues/1994
- https://github.com/getcursor/cursor/issues/1744
- https://github.com/getcursor/cursor/issues/1726
- https://github.com/getcursor/cursor/issues/1479
- https://github.com/getcursor/cursor/issues/1163
- https://github.com/getcursor/cursor/issues/931
- https://forum.cursor.com/t/document-indexing-settings/682/3
- https://forum.cursor.com/t/documentation-indexing-problems/20860/15
- https://forum.cursor.com/t/doc-sets-only-seem-to-index-a-single-page/32605

**Comment by NatElkins** [comment]
This is affecting me as well, only the entrypoint page is indexed.  Would love to see some progress on this!
